storage/src/main/java/org/apache/uniffle/storage/common/LocalStorage.java

```diff
@@ -81,9 +81,16 @@ private LocalStorage(Builder builder) {
       if (isEmptyAndWritableDir(baseFolder)) {
         LOG.warn("Base directory is already an empty dir, skip init");
       } else {
-        FileUtils.deleteDirectory(baseFolder);
-        if (!baseFolder.mkdirs()) {
-          throw new IOException("Failed to create base folder: " + basePath);
+        // base folder could be the mounted path, which could not be deleted.
+        // just clean the directory instead.
+        if (baseFolder.isDirectory()) {
```

Comment:
Concat is to line 83, i.e. `} else if () {`.
Maybe you should check `isDirectory` before `isEmptyAndWritableDir`.